flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendDynamicPropertiesTest.java

```diff
@@ -182,4 +196,45 @@ protected void executeProgram(Configuration configuration, PackagedProgram progr
                     program.getUserCodeClassLoader().getClass().getName());
         }
     }
+
+    public static void verifyCliFrontendWithDynamicProperties(
+            Configuration configuration,
+            String[] parameters,
+            GenericCLI cliUnderTest,
+            Duration expectedClientTimeout,
+            int expectedDefaultParallelism)
+            throws Exception {
+        TestingCliFrontendWithDynamicProperties testFrontend =
+                new TestingCliFrontendWithDynamicProperties(
+                        configuration,
+                        cliUnderTest,
+                        expectedClientTimeout,
+                        expectedDefaultParallelism);
+        testFrontend.run(parameters); // verifies the expected values (see below)
+    }
+
+    private static final class TestingCliFrontendWithDynamicProperties extends CliFrontend {
```

Comment:
I would like to have a more general testing class. Something like following.

```
    private static final class TestingCliFrontendWithDynamicProperties extends CliFrontend {
        private final Map<String, String> expectedConfigValues;

        ... ...

        @Override
        protected void executeProgram(Configuration configuration, PackagedProgram program) {
            expectedConfigValues.forEach(
                    (key, value) -> assertEquals(value, configuration.toMap().get(key)));
        }
    }
```

Reply:
@wangyang0918 ,
Thank you very much for reviewing. I will update the PR.